src/tests/unit_x509.cpp

```diff
@@ -1514,6 +1514,54 @@ Test::Result test_hashes(const Botan::Private_Key& key, const std::string& hash_
    return result;
 }
 
+Test::Result test_x509_tn_auth_list_extension_decode() {
+   /* cert with TNAuthList extension data was generated by asn1parse cfg:
+
+      asn1=SEQUENCE:tn_auth_list
+
+      [tn_auth_list]
+      spc=EXP:0,IA5:1001
+      range=EXP:1,SEQUENCE:TelephoneNumberRange
+      one=EXP:2,IA5:333
+
+      [TelephoneNumberRange]
+      start1=IA5:111
+      count1=INT:128
+      start2=IA5:222
+      count2=INT:256
+    */
+   const std::string filename("TNAuthList.pem");
+   Test::Result result("X509 TNAuthList decode");
+   result.start_timer();
+
+   Botan::X509_Certificate cert(Test::data_file("x509/x509test/" + filename));
+
+   using Botan::Cert_Extension::TNAuthList;
+
+   auto tn_auth_list = cert.v3_extensions().get_extension_object_as<TNAuthList>();
+
+   auto& tn_entries = tn_auth_list->entries();
+
+   result.confirm("cert has TNAuthList extension", tn_auth_list != nullptr, true);
+
+   result.test_eq("spc entry type", tn_entries[0].type() == TNAuthList::Entry::TN_ServiceProviderCode, true);
+   result.test_eq("spc entry data", tn_entries[0].service_provider_code(), "1001");
```

Comment:
Perhaps also test that the other types throw an error. For instance:

```C++
result.test_throws("SPC is no phone number", [&] { tn_entries[0].telephone_number(); });
```

Reply:
added throw tests for all 3 accessors